compatibility.js

```diff
@@ -69,6 +69,41 @@ const compatFunctions = {
         }
 
         return v2Config
+    },
+    v3: (unmodifiedConfig, config) => {
+        // Breaking changes: none, reasons stripped starting in v4
+
+        const v3Config = JSON.parse(JSON.stringify(config))
+
+        // Replace reasons
+        for (const key of Object.keys(unmodifiedConfig.features)) {
+            for (const i in unmodifiedConfig.features[key].exceptions) {
+                v3Config.features[key].exceptions[i].reason = unmodifiedConfig.features[key].exceptions[i].reason
+            }
+
+            if (key === 'trackerAllowlist') {
+                for (const domain of Object.keys(unmodifiedConfig.features[key].settings.allowlistedTrackers)) {
+                    for (const i in unmodifiedConfig.features[key].settings.allowlistedTrackers[domain].rules) {
+                        v3Config.features[key].settings.allowlistedTrackers[domain].rules[i].reason = unmodifiedConfig.features[key].settings.allowlistedTrackers[domain].rules[i].reason
+                    }
```

Comment:
Untested but you could reduce the level of boilerplate like this:

```suggestion
                    const rules = v3Config.features[key].settings.allowlistedTrackers[domain].rules
                    rules = rules.map((rule, i) => {
                        const reason = unmodifiedConfig.features[key].settings.allowlistedTrackers[domain].rules[i].reason
                        return Object.assign(rule, { reason })
                    })
```
This reduces the risk of `i`'s and `domains` and whatever being in the wrong order.

The latter part could be a shared function between the other rules potentially too.

Reply:
Also we could consider making sure reason is present here too.

```js
                    const rules = v3Config.features[key].settings.allowlistedTrackers[domain].rules
                    rules = rules.map((rule, i) => {
                        const reason = unmodifiedConfig?.features?.[key]?.settings?.allowlistedTrackers?.[domain]?.rules?.[i].reason || ""
                        return Object.assign(rule, { reason })
                    })
```